### PR TITLE
Add documentation for zero-copy locales

### PR DESCRIPTION
**Pull request by sffc**
Documents the conclusions from #831 

I put the docs in an empty module.

### CODE REVIEW

**Comment by sffc** [comment]
The docs will be available for preview here:

https://icu4x-pr-artifacts.storage.googleapis.com/gha/079d825b30abfd08849fb8efffb3e75ef3c5dbf1/docs/icu_locid/zerovec/index.html

**Review by Manishearth — APPROVED** [review]

File: components/locid/src/zerovec.rs [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+//!
+//! There are two main use cases, which have different solutions:
+//!
+//! 1. **Lookup:** You need to locate a locale in a zero-copy vector, such as when querying a map.
+//! 2. **Obtain:** You have a locale stored in a zero-copy vector, and you need to obtain a proper
+//!    [`Locale`] or [`LanguageIdentifier`] for use elsewhere in your program.
+//!
+//! # Lookup
+//!
+//! To perform lookup, store the BCP-47 byte array for the locale, and then use
```
Comment (zbraniecki) [inline components/locid/src/zerovec.rs:19]: `suggestion (n-b)`: `store the stringified locale in a canonical BCP-47 form as a byte array, and...`

File: components/locid/src/zerovec.rs [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+//! ];
+//! let zm: ZeroMap<u32, (Language, Option<Script>, Option<Region>)> =
+//!     data.iter().copied().collect();
+//!
+//! // Construct a LanguageIdentifier from a tuple entry
+//! let value = zm.get_copied(&25).expect("element is present");
+//! let mut lid = LanguageIdentifier::default();
+//! lid.language = value.0;
+//! lid.script = value.1;
+//! lid.region = value.2;
```
Comment (zbraniecki) [inline components/locid/src/zerovec.rs:79]: `suggestion (n-b)` it might be useful to add a convenience constructor `From<(Language, Script, Region)>`.

**Review by zbraniecki — APPROVED** [review]
lgtm!

File: components/locid/src/zerovec.rs [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+//! use icu_locid::langid;
+//! use zerovec::ZeroMap;
+//!
+//! // ZeroMap from integer to locale string
+//! let data: &[(u32, &[u8])] = &[
+//!     (5, b"de-DE-u-hc-h12"),
+//!     (10, b"en-US-u-ca-buddhist"),
+//!     (15, b"my-MM"),
+//!     (20, b"sr-Cyrl-ME"),
+//!     (25, b"zh-TW"),
```
Comment (robertbastian) [inline components/locid/src/zerovec.rs:105]: Maybe add an invalid example where parsing fails?

File: components/locid/src/zerovec.rs (hunk 2 above)
Comment (robertbastian) [reply components/locid/src/zerovec.rs:79]: +1

Comment (zbraniecki) [reply components/locid/src/zerovec.rs:79]: Spun off to https://github.com/unicode-org/icu4x/pull/1753

File: components/locid/src/zerovec.rs (hunk 3 above)
Comment (sffc) [reply components/locid/src/zerovec.rs:103]: Done.

File: components/locid/src/zerovec.rs (hunk 2 above)
Comment (sffc) [reply components/locid/src/zerovec.rs]: Done.

File: components/locid/src/zerovec.rs (hunk 1 above)
Comment (sffc) [reply components/locid/src/zerovec.rs]: Done.

**Review by Manishearth — APPROVED** [review]

**Pull request merged**